### PR TITLE
Adding envsetup to boostrap development env

### PR DESCRIPTION
**Pull request by anduonghien**
https://github.com/nimblehq/elixir-templates/issues/8

## What happened

Adding the **Makefile** and **docker-compose.dev.yml** to help new developer could bootstrap the development env easily
 
## Insight

Just using docker-compose.
 
## Proof Of Work

1/ Start docker container

image

2/ mix test

image

3/ mix ecto.create

image

### CODE REVIEW

**Comment by anduonghien** [comment]
> Good to add Docker Compose.
> 
> As for bootrapping the app, since by default there is no `bin` folder (contrary to Rails), I think we should use a Makefile instead. It's more standard and would not introduce a non-typical directory.

I have changed it to Makefile in f3b89c6 @olivierobert 

image

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version: "3.2"
+
+services:
+  db:
+    image: postgres:12.3
+    container_name: Nimble
+    environment:
+      - POSTGRES_HOST_AUTH_METHOD=trust
+      - POSTGRES_DB=Nimble_development
```
Comment (Tolsee) [inline docker-compose.dev.yml:9]: 🤔 
```suggestion
      - POSTGRES_DB=nimble_dev
```

Comment (anduonghien) [reply docker-compose.dev.yml:9]: Sure, will change it @Tolsee

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.PHONY: env_setup
+
+env_setup: ## Setup development environment
```
Comment (olivierobert) [inline Makefile:3]: How about simply `setup`?

Comment (anduonghien) [reply Makefile:3]: I prefer to keep it as `env_setup` as the `setup` is bigger than the `env`, maybe later we have another `setup step` so we could separate it to different `step`, and if we want to combine that into 1 setup step, Makefile allow us to do that

```
setup: env_setup another_setup

env_setup:
 - xxx

another_setup:
 - yyy
```

Comment (olivierobert) [reply Makefile:3]: To me, it feels like inheriting from the Rails project. And actually there is no environment setup at all, just the db 😆 How about `docker_setup` then? or `db_setup`?

Comment (anduonghien) [reply Makefile:3]: We can go with `docker_setup` @olivierobert

**Comment by Tolsee** [comment]
> What happened
Adding the bin/envsetup.sh and docker-compose.dev.yml to help new developer could boostrap the development env easily

Please update the PR description

File: docker-compose.dev.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+version: "3.2"
+
+services:
+  db:
+    image: postgres:12.3
+    container_name: nimble
```
Comment (rosle) [inline docker-compose.dev.yml:6]: Should we add `_db` suffix 📝

Comment (anduonghien) [reply docker-compose.dev.yml]: Fixed that in https://github.com/nimblehq/elixir-templates/pull/9/commits/3be8913e8e8017ff7d6ebf7e52b22266adef3dd0 @rosle

**Pull request merged**